### ISSUE TITLE
Connection Leak

### ISSUE THREAD

**Issue by lefthandedgoat** [fsprojects/FSharp.Data.SqlClient#174]
I could not find any previous related issues.  I have repro steps an a work around and I think I know what the source of the problem is.

```
open FSharp.Data

[<Literal>]
let connectionString = """Server=localhost\sqlexpress;User Id=sa;Password=123123;max pool size = 10"""

[<Literal>]
let someQuery = """SELECT TOP 1 * from testdb.dbo.testtable"""
type SomeQuery = SqlCommandProvider<someQuery, connectionString>
let runQuery () =
  use cmd = new SomeQuery(connectionString)
  cmd.Execute()
  
let results = 
  [1..1000]
  |> List.iter (fun i ->
    printfn "iteration %i" i 
    runQuery() |> ignore)
 
System.Console.ReadKey() |> ignore   
```

This will fail after the 10th query (max pool size is 10)

Changing
```runQuery() |> ignore```
to
```runQuery() |> Seq.head |> ignore```

Fixes the problem.

We are ignoring the results of the cmd.Execute because its a stored proc that modifies things and we do not care what the results are.

I think that this may be the root of the issue?
https://github.com/fsprojects/FSharp.Data.SqlClient/blob/91147eabbf6e9e174b8cd1c6e0619c18c8386766/src/SqlClient/SqlClientExtensions.fs#L27-L31

I suspect that the results of the command are never disposed if the `seq { ...  }` expression is never iterated over.

Its no rush for me as I have a reasonable work around.

**Comment by dmitry-a-morozov** [comment]
Thanks for reporting that. 
Just curious: does the stored procedure return a result set in addition to modifications? 
Because if it's only data updates a result type suppose to be of type<int>.

**Comment by lefthandedgoat** [comment]
Due to the other issue I reported (which I still havent had a chance to look into), the proc does return a bogus result set at the top, to prevent the built in sql procedure from traversing further to find out what the real result set is and discovering a circular reference =).

**Comment by dmitry-a-morozov** [comment]
fixed in v1.8
